libraries/pod/src/non_zero_pubkeys.rs

```diff
@@ -0,0 +1,344 @@
+    }
+}
+impl TryFrom<Option<ElGamalPubkey>> for OptionalNonZeroElGamalPubkey {
+    type Error = ProgramError;
+    fn try_from(p: Option<ElGamalPubkey>) -> Result<Self, Self::Error> {
+        match p {
+            None => Ok(Self(ElGamalPubkey::default())),
+            Some(elgamal_pubkey) => {
+                if elgamal_pubkey == ElGamalPubkey::default() {
+                    Err(ProgramError::InvalidArgument)
+                } else {
+                    Ok(Self(elgamal_pubkey))
+                }
+            }
+        }
+    }
+}
+impl From<OptionalNonZeroElGamalPubkey> for Option<ElGamalPubkey> {
+    fn from(p: OptionalNonZeroElGamalPubkey) -> Self {
+        if p.0 == ElGamalPubkey::default() {
+            None
+        } else {
+            Some(p.0)
+        }
+    }
+}
+
+#[cfg(any(feature = "serde-traits", test))]
+const OPTIONAL_NONZERO_ELGAMAL_PUBKEY_LEN: usize = 32;
+
+#[cfg(feature = "serde-traits")]
+impl Serialize for OptionalNonZeroElGamalPubkey {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if self.0 == ElGamalPubkey::default() {
+            s.serialize_none()
+        } else {
+            s.serialize_some(&self.0.to_string())
+        }
+    }
+}
+
+#[cfg(feature = "serde-traits")]
+pub(crate) struct OptionalNonZeroElGamalPubkeyVisitor;
+
+#[cfg(feature = "serde-traits")]
+impl<'de> Visitor<'de> for OptionalNonZeroElGamalPubkeyVisitor {
+    type Value = OptionalNonZeroElGamalPubkey;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("an ElGamal public key as base64 or `null`")
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        let bytes = BASE64_STANDARD.decode(v).map_err(Error::custom)?;
+
+        if bytes.len() != OPTIONAL_NONZERO_ELGAMAL_PUBKEY_LEN {
+            return Err(Error::custom(format!(
+                "Length of base64 decoded bytes is not {}",
+                OPTIONAL_NONZERO_ELGAMAL_PUBKEY_LEN
+            )));
+        }
+
+        let mut array = [0; OPTIONAL_NONZERO_ELGAMAL_PUBKEY_LEN];
+        array.copy_from_slice(&bytes[0..OPTIONAL_NONZERO_ELGAMAL_PUBKEY_LEN]);
+        let elgamal_pubkey = ElGamalPubkey(array);
+        OptionalNonZeroElGamalPubkey::try_from(Some(elgamal_pubkey)).map_err(Error::custom)
+    }
+
+    fn visit_unit<E>(self) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        Ok(OptionalNonZeroElGamalPubkey::default())
+    }
+}
+
+#[cfg(feature = "serde-traits")]
+impl<'de> Deserialize<'de> for OptionalNonZeroElGamalPubkey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_any(OptionalNonZeroElGamalPubkeyVisitor)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {super::*, crate::utils::pod_from_bytes};
+
+    const OPTIONAL_NONZERO_PUBKEY_LEN: usize = 32;
```

Comment:
nit: you can just use https://github.com/solana-labs/solana/blob/37887d487ce9b87297d9a20dd07a89adf4c66cd5/sdk/program/src/pubkey.rs#L18